### PR TITLE
[CI] Add unit tests for srt/utils/auth.py

### PR DESCRIPTION
**Pull request by Lidang-Jiang**
## Summary

- Add 29 unit tests for `srt/utils/auth.py` (auth decision logic)
- Contributes to #20865 (Improve Unit Test Coverage)
- Tests are CPU-only, no server, no model loading, no FastAPI required

**Test coverage for `decide_request_auth` (pure function):**

| Auth Level | Scenarios Tested |
|-----------|-----------------|
| Always-allowed | OPTIONS method, /health/*, /metrics |
| NORMAL | no keys, correct key, wrong key, missing header, admin-only config |
| ADMIN_FORCE | no admin key (→403), correct admin key, wrong key, api_key rejected |
| ADMIN_OPTIONAL | no keys, api_key only, admin_key only, both keys (requires admin) |
| Edge cases | malformed Bearer, empty header, case-insensitive Bearer |

Also tests `AuthDecision` (frozen dataclass), `AuthLevel` enum, and `@auth_level` decorator.

## Test plan

<details>
<summary>Before / After test run comparison</summary>

**Before (on main) — 4 test files in utils/, 34 passed:**
```
test/registered/unit/utils/
├── test_gauge_histogram.py
├── test_json_response.py
├── test_patch_tokenizer.py
└── test_profile_merger.py

34 passed, 4 warnings in 60.01s
```

**After (this PR) — 5 test files, 29 new tests all pass:**
```
test/registered/unit/utils/
├── test_auth.py                    ← NEW (29 tests)
├── test_gauge_histogram.py
├── test_json_response.py
├── test_patch_tokenizer.py
└── test_profile_merger.py

29 passed, 1 warning in 4.33s (new tests only)
```

</details>

- [x] All 29 new tests pass locally
- [x] Existing 34 utils tests still pass (no regression)
- [x] Follows `test/registered/unit/README.md` conventions
- [x] Registered for CPU CI via `register_cpu_ci`
- [x] No torch/FastAPI/model imports — lightweight as intended by module design

### CODE REVIEW

File: test/registered/unit/utils/test_auth.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+
+class TestAuthLevel(unittest.TestCase):
+    def test_enum_values(self):
+        self.assertEqual(AuthLevel.NORMAL.value, "normal")
+        self.assertEqual(AuthLevel.ADMIN_OPTIONAL.value, "admin_optional")
+        self.assertEqual(AuthLevel.ADMIN_FORCE.value, "admin_force")
+
+    def test_is_string_enum(self):
+        self.assertIsInstance(AuthLevel.NORMAL, str)
```
[the remark this answers is not in the document]
Comment (Lidang-Jiang) [reply test/registered/unit/utils/test_auth.py:41]: Good catch — `str(AuthLevel.NORMAL)` behavior varies between Python versions (`str, Enum` vs 3.11+ changes). Fixed in 6efd96c63: switched to `assertEqual(AuthLevel.NORMAL, "normal")` which works on all versions since `str, Enum` allows direct string comparison. Also switched all test classes to `CustomTestCase` per project convention.

**Review by ispobock — APPROVED** [review]
